pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +71,86 @@ bc:
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.ErrorSeverity,
+						Key:      MissingInputImageStreamErr,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("create the image stream %s with the required images", imageStreamNode.ResourceString())),
+					})
+
+					continue OUTER
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
+						markers = append(markers, osgraph.Marker{
+							Node: bcNode,
+							RelatedNodes: []graph.Node{istNode,
+								imageStreamNode},
+
+							Severity: osgraph.WarningSeverity,
+							Key:      MissingImageStreamTagWarning,
+							Message: fmt.Sprintf("%s builds from %s, but the image stream tag does not exist.",
+								bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamTagNode).ResourceString()),
+							Suggestion: osgraph.Suggestion(fmt.Sprintf("examine analysis of build config outputs from this command and see if they build %s", istNode.(*imagegraph.ImageStreamTagNode).ResourceString())),
+						})
+					}
+					continue OUTER
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamImageNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					// cycle through the imageStream.Status.Tags.[]TagEvent.DockerImageReference and see if the image id is present
+					var imageID string
+					for _, tagEventList := range imageStream.Status.Tags {
+						dockerImageReference, err := imageapi.ParseDockerImageReference(tagNode.Name)
+						if err == nil {
+							for _, tagEvent := range tagEventList.Items {
+								if strings.Contains(tagEvent.DockerImageReference, dockerImageReference.ID) {
+									break OUTER
+								}
+							}
+						}
+					}
+					// if got here, then we did not find the ref
+					markers = append(markers, osgraph.Marker{
+						Node: bcNode,
+						RelatedNodes: []graph.Node{istNode,
+							imageStreamNode},
+
+						Severity: osgraph.WarningSeverity,
+						Key:      MissingImageStreamTagWarning,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream image does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamImageNode).ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("tag the image with hexadecimal ID %s into the image stream %s", imageID, imageStreamNode.ResourceString())),
```

Comment:
Before I go too far down a path that is wrong, are you referring to a 1 time cycle through the image streams where I check the annotations similar to https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/importimage.go#L171 ?